pkg/userdata/bottlerocket/prepare.go

```diff
@@ -28,6 +30,22 @@ func patchCertPaths(input *userdata.BaseUserData) {
 	}
 }
 
+func buildEtcdadmArgs(config bootstrapv1alpha3.BottlerocketConfig) userdata.EtcdadmArgs {
+	repository, tag := splitRepositoryAndTag(config.EtcdImage)
+	return userdata.EtcdadmArgs{
+		Version:         strings.TrimPrefix(tag, "v"), // trim "v" to get pure simver because that's what etcdadm expects.
+		ImageRepository: repository,
+	}
+}
+
+func splitRepositoryAndTag(image string) (repository, tag string) {
+	lastInd := strings.LastIndex(image, ":")
+	if lastInd == -1 || lastInd == len(image)-1 {
```

Comment:
the first part of if condition where `:` is not a part of image at all will return the image repo correctly, but the second part that checks if the `:` was at the end will return image containing the `:` at the end, so I think it should be:
```suggestion
	if lastInd == -1 {
		return image, ""
	}
	if lastInd == len(image)-1 {
		return image[:lastInd], ""
	}
```